threescale/types.go

```diff
@@ -59,7 +59,7 @@ type AuthorizeResponse struct {
 
 // AuthType maps to a known client authentication pattern
 // Currently known and supported are 0=ServiceToken 1=ProviderKey
-type AuthType int
+type AuthType string
```

Comment:
The problem is the conversion to query params. How would we offer that? So a user passes a given iota and that needs to translate somehow to a string. When we are providing a fixed value up front that work but not in the case where we can't map the iota to a value right?

I think moving to string is fine and simplifies it but open to suggestion for sure.

Reply:
It is fine, I was thinking more in line with the client knowing it is not using any of the known authentication types, but perhaps no code depends on it (ie. some optional validation could ensure the known types validate against known formats, but since it is now out it is probably not of use).

Reply:
Yes, well they will know because they will need to cast their string as an AuthType to use anything outside `ServiceToken` and `ProviderKey`